### PR TITLE
Lc problem15 letter combinations of a phone number

### PR DESCRIPTION
**Pull request by CHAriina**
solving the letter-combinations-of-a-phone-number problem (day 15 first item)

### CODE REVIEW

File: ScnSoft-Education/LeetCodeTraining.Tests/LetterCombinationsOfAPhoneNumber.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        private readonly Solution resultSolution;
+            
+        public SolutionTests()
+        {
+            resultSolution = new Solution();
+        }
+
+        [Theory]
+        [InlineData("23", new string[] { "ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf" })]
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining.Tests/LetterCombinationsOfAPhoneNumber.Tests.cs:15]: add unit test for "1"

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                { '5', "jkl" },
+                { '6', "mno" },
+                { '7', "pqrs" },
+                { '8', "tuv" },
+                { '9', "wxyz" }
+            };
+
+            var res = new List<string>();
+            if (string.IsNullOrEmpty(digits))
+                return res;
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs:22]: add braces ( {..} ) after if condition

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+            res.Add(string.Empty);
+            for (var i = 0; i < digits.Length; i++)
+                res = res
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs]: add braces ( {..} ) after for

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                    .SelectMany(x => digitsLettersMap[digits[i]].Select(y => x + y.ToString()))
+                    .ToList();
+
+            return res;
+
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs:36]: remove empty line

File: ScnSoft-Education/LeetCodeTraining.Tests/LetterCombinationsOfAPhoneNumber.Tests.cs (hunk 1 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining.Tests/LetterCombinationsOfAPhoneNumber.Tests.cs:15]: added test for 1

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs (hunk 2 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs:22]: added {}

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs (hunk 3 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs]: added {}

File: ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs (hunk 4 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/LetterCombinationsOfAPhoneNumber15/LetterCombinationsOfAPhoneNumber.cs:36]: deleted empty line

**Pull request merged**